### PR TITLE
Inflections for acronyms, rename classes accordingly

### PR DESCRIPTION
**Pull request by nimmolo**
Note: the Zeitwerk PR had many parts, so i'm breaking them up. This is the first. Feel free to suggest I break them up further. 

Background: Zeitwerk is less magical, more straightforward, than the "classic" Rails autoloader. (The autoloader is what makes it unnecessary, during development, to explicitly require every class everywhere every time you change code. It has a configured scope: for example it will not reload anything in `app/lib` by default.) "Classic" did a lot of inferring of possible intent, checking filenames, declared class names, module names, etc. in order to keep programmers from having to follow too many conventions.

Zeitwerk is "dumber" but more efficient and faster: it enforces several rules and ignores everything not in its scope. 
- Folder scope is the outer constraint. We can tell it to ignore folders; nothing inside will be autoloaded. It will attempt to autload everything inside, period.
- Naming is strict. The class or module name must be the camelized constantized version of the snake case filename, filtered only by Rails (or a subset, autoloader) inflections.
- One top level class per file(!) I believe we can load multiple classes inside a module, however.

Because Inflections play such a big part in Zeitwerk's expectation of class name from filename, I've tried to keep the inflections as dumb as possible. This is a choice: whereas we need `API2` to be inflected as such to match the class, the base inflector will try to match any "word" in snake case to that pattern, so it would look for `api_helper.rb` > `APIHelper`. To avoid renaming classes, we could write special inflections for `ApiHelper` and `Api2Controller`, and many others like `BadApiKeyError` or whatever else we may write in the future. We'd have to keep adding inflections.

It seemed more foolproof to me to just fix these acronyms in all-caps, so we don't have to think about it, and change the class names to be consistent. 

On the other hand, I do kind of like the way `ApiHelper` looks. But now's the time to speak up — it's up to you guys.

### CODE REVIEW

File: app/models/api_key.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # frozen_string_literal: true
 
-class ApiKey < AbstractModel
+class APIKey < AbstractModel
```
[the remark this answers is not in the document]
Comment (pellaea) [reply app/models/api_key.rb:3]: These are false positives.  Rubocop can't know that I use a different but equivalent mechanism to ensure only allowed parameters are accepted and passed through to create!.

File: app/classes/api/name_api.rb [hunk 2]
```diff
@@ -263,7 +263,7 @@ def change_deprecated(name)
 
     def add_synonym(name)
       return unless @synonymize_with
-      raise(CanOnlySynonymizeUnsynonimizedNames.new) if name.synonym_id
+      raise(CanOnlySynonymizeUnsynonymizedNames.new) if name.synonym_id
```
Comment (JoeCohen) [inline app/classes/api/name_api.rb:266]: GitHub claims that this line is changed. I don't see a difference. Am I missing something?

Comment (JoeCohen) [reply app/classes/api/name_api.rb:266]: Yes. I missed the spelling correction of Unsynon<ins>**i**</ins>mize to Unsynon<ins>**y**</ins>mize

**Comment by JoeCohen** [comment]
I will change this to a Draft PR so that I don't accidentally merge it before we've had a chance to discuss it at a Tech meeting.

File: app/models/api_key.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (nimmolo) [reply app/models/api_key.rb:3]: I like that idea, Joe!

**Comment by nimmolo** [comment]
@JoeCohen — Agree about waiting til the meeting. 
A prerequisite to implementing this inflection, with minimum fuss, would be retiring the API.

File: app/models/api_key.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (nimmolo) [reply app/models/api_key.rb:3]: Possible replacement: https://github.com/marketplace/actions/brakeman-linter

**Comment by nimmolo** [comment]
Closing this PR since it necessarily changes API. 
Everything will be in a single Zeitwerk PR.

**Pull request closed without merging**